capstone/src/main/java/com/google/step/clients/ScoreClient.java

```diff
@@ -0,0 +1,163 @@
+package com.google.step.clients;
+
+import com.google.step.data.RestaurantPageViews;
+import com.google.step.data.RestaurantScore;
+import com.google.step.data.WeeklyPageView;
+
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Map;
+import java.util.HashMap;
+import java.util.Collections;
+
+
+/**
+ * Will take care of calculating the score for each restaurant.
+ */
+public class ScoreClient {
+
+    private final float minimum = 10.0f;
+    private final MetricsClient metricsClient = new MetricsClient();
+
+    /**
+     * Only method to be called and calculates the scores of each restaurant.
+     * @return an unmodifiable list of restaurantScore class.
+     */
+    public List<RestaurantScore> calculateScores() {
+        Map<Long, Double> scoreMap = new HashMap<>();
```

Comment:
Why is this a Map<Long, Double> instead of a list?

Reply:
In order to keep track of the restaurant score at the same time. That way, by also having a list with the id's I can iterate easier through the map when standardizing and normalizing the score.

Reply:
I may be missing it, but I don't don't see any place where you use the map to get a score for a specific key (other than iterating over every key in the map). 

If you only ever iterate over every restaurant/score pair, then this could be a list (or even a set) of items (you could create a class to store the long and double values). A list would be more space efficient (under the hood the map uses an array whose length is greater than the number of keys stored in the map). 
 
Even if you decide to stay with the map, there is no need to have a separate list with the ids. You can iterate over the map using the entry set method and a for each loop: https://www.geeksforgeeks.org/iterate-map-java/

Reply:
I basically changed everything to use lists and should be a lot quicker now! I'm no longer using maps at all.